### ISSUE TITLE
cannot find libstdc++ headers

### ISSUE THREAD

**Issue by mgood7123** [tpoechtrager/osxcross#160]
$ target/bin/o32-clang
osxcross: error: cannot find libstdc++ headers
osxcross: error: while detecting target

compiled with

`
#!/usr/bin/env bash
#
# Package the OS X SDKs into a tar file to be used by `build.sh`.
#

export LC_ALL=C

function set_xcode_dir()
{
  local tmp=$(ls $1 2>/dev/null | grep "^Xcode.*.app" | grep -v "beta" | head -n1)

  if [ -z "$tmp" ]; then
    tmp=$(ls $1 2>/dev/null | grep "^Xcode.*.app" | head -n1)
  fi

  if [ -n "$tmp" ]; then
    XCODEDIR="$1/$tmp"
  fi
}

if [ $(uname -s) != "Darwin" ]; then
  if [ -z "$XCODEDIR" ]; then
    echo "This script must be run on OS X" 1>&2
    echo "... Or with XCODEDIR=... on Linux" 1>&2
    exit 1
  else
    case $XCODEDIR in
      /*) ;;
      *) XCODEDIR="$PWD/$XCODEDIR" ;;
    esac
    set_xcode_dir $XCODEDIR
  fi
else
  set_xcode_dir $(echo /Volumes/Xcode* | tr ' ' '\n' | grep -v "beta" | head -n1)

  if [ -z "$XCODEDIR" ]; then
    set_xcode_dir /Applications

    if [ -z "$XCODEDIR" ]; then
      set_xcode_dir $(echo /Volumes/Xcode* | tr ' ' '\n' | head -n1)

      if [ -z "$XCODEDIR" ]; then
        echo "please mount Xcode.dmg" 1>&2
        exit 1
      fi
    fi
  fi
fi

if [ ! -d $XCODEDIR ]; then
  echo "cannot find Xcode (XCODEDIR=$XCODEDIR)" 1>&2
  exit 1
fi

echo -e "found Xcode: $XCODEDIR"

WDIR=$(pwd)

which gnutar &>/dev/null

if [ $? -eq 0 ]; then
  TAR=gnutar
else
  TAR=tar
fi

which xz &>/dev/null

if [ $? -eq 0 ]; then
  COMPRESSOR=xz
  PKGEXT="tar.xz"
else
  COMPRESSOR=bzip2
  PKGEXT="tar.bz2"
fi

set -e

pushd $XCODEDIR &>/dev/null

if [ -d "Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs" ]; then
  pushd "Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs" &>/dev/null
else
  if [ -d "../Packages" ]; then
    pushd "../Packages" &>/dev/null
  elif [ -d "Packages" ]; then
    pushd "Packages" &>/dev/null
  else
    if [ $? -ne 0 ]; then
      echo "Xcode (or this script) is out of date" 1>&2
      echo "trying some magic to find the SDKs anyway ..." 1>&2

      SDKDIR=$(find . -name SDKs -type d | grep MacOSX | head -n1)

      if [ -z "$SDKDIR" ]; then
        echo "cannot find SDKs!" 1>&2
        exit 1
      fi

      pushd $SDKDIR &>/dev/null
    fi
  fi
fi

SDKS=$(ls | grep "^MacOSX10.*" | grep -v "Patch")

if [ -z "$SDKS" ]; then
    echo "No SDK found" 1>&2
    exit 1
fi

# Xcode 5
LIBCXXDIR1="Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/c++/v1"

# Xcode 6
LIBCXXDIR2="Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1"

# Manual directory
MANDIR="Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/share/man"
declare -p SDKS
declare -p WDIR
for SDK in $SDKS; do
  echo -n "packaging $(echo "$SDK" | sed -E "s/(.sdk|.pkg)//g") SDK "
  echo "(this may take several minutes) ..."

  if [[ $SDK == *.pkg ]]; then
    cp $SDK $WDIR
    continue
  fi
  TMP=$(mktemp -d /tmp/XXXXXXXXXXX)
  mkdir $WDIR/$(readlink $SDK)
  ln -sv $WDIR/$(readlink $SDK) $TMP/$SDK

  pushd $XCODEDIR &>/dev/null

  # libc++ headers for C++11/C++14
  mkdir -p $TMP/$SDK/usr/include/c++
  if [ -d $LIBCXXDIR1 ]; then
    cp -rf $LIBCXXDIR1 "$TMP/$SDK/usr/include/c++"
  elif [ -d $LIBCXXDIR2 ]; then
    cp -rf $LIBCXXDIR2 "$TMP/$SDK/usr/include/c++"
  fi

  if [ -d $MANDIR ]; then
    mkdir -p $TMP/$SDK/usr/share/man
    cp -rf $MANDIR/* $TMP/$SDK/usr/share/man
  fi

  popd &>/dev/null

  pushd $TMP &>/dev/null
  $TAR -cf - * | $COMPRESSOR -9 -c - > "$WDIR/$SDK.$PKGEXT"
  popd &>/dev/null

  rm -rf $TMP
done

popd &>/dev/null
popd &>/dev/null

echo ""
ls -lh | grep MacOSX
`

and Xcode_10_Beta_6.xip SDK

**Comment by mgood7123** [comment]
it appears to default to --stdlib=libstdc++

**Comment by mgood7123** [comment]
as -stdlib=libc++ works (but is untested)

**Comment by mgood7123** [comment]
testing o32-clang ... /home/mobile_c/git/ccr_resources/diss/apple/osxcross/oclang/test.c:1:10: fatal error: 'stdio.h' file not found
#include <stdio.h>

**Comment by jay-hankins** [comment]
Seeing this as well in macOS 10.14:
```
root@72bcd326f3f6:/osxcross# CMAKE_CXX_COMPILER=i386-apple-darwin15-clang++-libc++ OSX_VERSION_MIN=10.12 ./build.sh

extracting MacOSX10.14.sdk.tar.xz ...
compiling wrapper ...

installing CMake
testing o32-clang ... osxcross: error: cannot find libc++ headers
osxcross: error: while detecting target

exiting with abnormal exit code (1)
```

What's going on with this, is there any way I can help? What does @melvinvermeeren's commit change?

**Comment by tpoechtrager** [comment]
Your SDK does not contain libc++ headers. Follow the README to package the SDK properly.

**Comment by jay-hankins** [comment]
@tpoechtrager I followed these steps completely:

> Packaging the SDK on Mac OS X:
> 1. [Download Xcode **]
> 2. [Mount Xcode.dmg (Open With -> DiskImageMounter) ***]
> 3. Run: ./tools/gen_sdk_package.sh (from the OSXCross package)
> 4. Copy the packaged SDK (*.tar.* or *.pkg) on a USB Stick
> 5. (On Linux/BSD) Copy or move the SDK into the tarballs/ directory of OSXCross.

Are there additional steps to make the SDK package libc++ headers, aside from running `gen_sdk_package.sh`?

**Comment by tpoechtrager** [comment]
What version of Xcode are you using? Up to 7.3.x is supported by the script.

**Comment by jay-hankins** [comment]
@tpoechtrager Yes, that fixed it! My program compiled successfully and I'm thrilled! 😄 

I actually had seen the requirement for Xcode 7.3.x but then had trouble locating the download on Apple's site, and by the time I got around to building osxcross, had forgotten about locating the old version.
